### PR TITLE
James 2813 bind task serialization

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/TaskSerializationModule.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import org.apache.mailbox.tools.indexer.ReIndexerPerformer;
+import org.apache.mailbox.tools.indexer.ReprocessingContextInformationDTO;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.multibindings.ProvidesIntoSet;
+
+public class TaskSerializationModule extends AbstractModule {
+
+    @Override
+    protected void configure() {
```
Comment (remk) [inline server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/TaskSerializationModule.java]: do we still nee to override it ?

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/TaskEventsSerializationTest.java [hunk 2]
```diff
@@ -83,10 +90,14 @@ void taskShouldBeDeserializable(TaskEvent event, String serializedJson) throws E
...
+            Arguments.of(new Completed(AGGREGATE_ID, EVENT_ID, Task.Result.PARTIAL, Option.apply(counterAdditionalInformation)), "{\"result\":\"PARTIAL\",\"aggregate\":\"2c7f4081-aa30-11e9-bf6c-2d3b9e84aafd\",\"event\":42,\"type\":\"task-manager-completed\",\"additionalInformation\":\"{\\\"type\\\":\\\"memory-reference-task-with-counter\\\",\\\"count\\\":3}\"}"),
+            Arguments.of(new Failed(AGGREGATE_ID, EVENT_ID, Option.apply(counterAdditionalInformation)), "{\"aggregate\":\"2c7f4081-aa30-11e9-bf6c-2d3b9e84aafd\",\"event\":42,\"type\":\"task-manager-failed\",\"additionalInformation\":\"{\\\"type\\\":\\\"memory-reference-task-with-counter\\\",\\\"count\\\":3}\"}"),
+            Arguments.of(new Cancelled(AGGREGATE_ID, EVENT_ID, Option.apply(counterAdditionalInformation)), "{\"aggregate\":\"2c7f4081-aa30-11e9-bf6c-2d3b9e84aafd\",\"event\":42,\"type\":\"task-manager-cancelled\",\"additionalInformation\":\"{\\\"type\\\":\\\"memory-reference-task-with-counter\\\",\\\"count\\\":3}\"}")
```
Comment (mbaechler) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/TaskEventsSerializationTest.java]: can't we get rid of this extra level of quoting?

Comment (rouazana) [reply server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/TaskEventsSerializationTest.java]: it's not in this PR, but in #2740

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java [hunk 3]
```diff
@@ -128,13 +129,18 @@
         new TikaMailboxModule(),
         new SpamAssassinListenerModule());
 
-    public static Module ALL_BUT_JMX_CASSANDRA_MODULE = Modules.combine(
+    public static Module REQUIRE_TASK_MANAGER_MODULE = Modules.combine(
```
[the remark this answers is not in the document]
Comment (Arsnael) [reply server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java]: Unfortunately I don't have a better idea for now... but thanks for the explanation

**Comment by rouazana** [comment]
timeout
```
[8cde0bed630832561a35de74f18a9700c3754a0d] [INFO] Apache James :: Server :: JMAP (draft) :: RabbitMQ + Object Store + Cassandra Integration testing FAILURE [  02:00 h]
```
test this please...

**Comment by rouazana** [comment]
```
[8cde0bed630832561a35de74f18a9700c3754a0d] [ERROR] Errors: 
[8cde0bed630832561a35de74f18a9700c3754a0d] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
```
test this please

**Comment by Arsnael** [comment]
```
[8cde0bed630832561a35de74f18a9700c3754a0d] 15:51:30.445 [ERROR] c.l.e.m.ElasticsearchReporter - Error adding metrics template to elasticsearch: 400/Bad Request
[8cde0bed630832561a35de74f18a9700c3754a0d] 15:51:35.089 [ERROR] c.l.e.m.ElasticsearchReporter - Error adding metrics template to elasticsearch: 400/Bad Request
```
hu? test this please

**Comment by rouazana** [comment]
> hu? test this please

The test class is timeouting... We are looking at it.

**Comment by remk** [comment]
```
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at java.base/java.lang.Thread.run(Thread.java:834)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 12:33:20.392 [ERROR] o.a.j.u.c.ConcurrentTestRunner - Error caught during concurrent testing (iteration 3421, threadNumber 1)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] java.lang.IllegalStateException: Event Bus is not running
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at org.apache.james.mailbox.events.RabbitMQEventBus.dispatch(RabbitMQEventBus.java:119)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at org.apache.james.mailbox.events.EventBus.dispatch(EventBus.java:56)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at org.apache.james.mailbox.events.RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.lambda$dispatchShouldStopDeliveringEventsShortlyAfterStopIsCalled$9(RabbitMQEventBusTest.java:570)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at org.apache.james.util.concurrency.ConcurrentTestRunner$ConcurrentRunnableTask.run(ConcurrentTestRunner.java:120)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] 	at java.base/java.lang.Thread.run(Thread.java:834)
[4ef60e77d98eea157ba97e7f9f568e4b474a94ca] [INFO] 
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**